docs/writing_a_simple_publisher_and_subscriber.md

```diff
@@ -0,0 +1,527 @@
+# Writing a simple publisher and subscriber (RUST)
+* Goal: Create and run a publisher and subscriber node using RUST.
+* Tutorial level: Beginner
+* Time: 20 minutes
+<details><summary>Background</summary>
+
+In this tutorial you will create a pair of 
+[nodes](https://docs.ros.org/en/humble/Tutorials/Beginner-CLI-Tools/Understanding-ROS2-Nodes/Understanding-ROS2-Nodes.html) that pass information to each other via a 
+[topic](https://docs.ros.org/en/humble/Tutorials/Beginner-CLI-Tools/Understanding-ROS2-Topics/Understanding-ROS2-Topics.html) in the form of string messages. The example used here is a simple "talker" and "listener" system; one node publishes data and the other subscribes to the topic to receive that data.
+
+Since RUST doesn't have inheritance, it's not possible to inherit from `Node` as is common practice in [`rclcpp`](https://docs.ros.org/en/humble/Tutorials/Beginner-Client-Libraries/Writing-A-Simple-Cpp-Publisher-And-Subscriber.html) or [`rclpy`](https://docs.ros.org/en/humble/Tutorials/Beginner-Client-Libraries/Writing-A-Simple-Py-Publisher-And-Subscriber.html).
+
+The code used in these examples can be found [here](https://github.com/Guelakais/ros2_rust/tree/better_tutorial/examples/your_package_name)  
```

Comment:
This will be invalid once this PR is merged, can you check if relative paths work? Otherwise, replace this with `main`

Reply:
done